vm/actor/src/builtin/power/mod.rs

```diff
@@ -583,127 +366,156 @@ impl Actor {
 
         Ok(())
     }
 
-    fn get_claim_or_abort<BS: BlockStore>(
-        st: &State,
-        store: &BS,
-        a: &Address,
-    ) -> Result<Claim, ActorError> {
-        st.get_claim(store, a)
-            .map_err(|e| {
+    fn submit_porep_for_bulk_verify<BS, RT>(
+        rt: &mut RT,
+        seal_info: SealVerifyInfo,
+    ) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_type(std::iter::once(&*MINER_ACTOR_CODE_ID))?;
+
+        rt.transaction::<State, _, _>(|st, rt| {
+            let mut mmap = if let Some(ref batch) = st.proof_validation_batch {
+                Multimap::from_root(rt.store(), batch).map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("failed to load proof batching set: {}", e),
+                    )
+                })?
+            } else {
+                Multimap::new(rt.store())
+            };
+
+            let miner_addr = rt.message().from();
+            mmap.add(miner_addr.to_bytes().into(), seal_info)
+                .map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("failed to insert proof into set: {}", e),
+                    )
+                })?;
+
+            let mmrc = mmap.root().map_err(|e| {
                 ActorError::new(
                     ExitCode::ErrIllegalState,
-                    format!("failed to load claim for miner {}: {}", a, e),
-                )
-            })?
-            .ok_or_else(|| {
-                ActorError::new(
-                    ExitCode::ErrIllegalArgument,
-                    format!("no claim for miner {}", a),
+                    format!("failed to flush proofs batch map: {}", e),
                 )
-            })
+            })?;
+            st.proof_validation_batch = Some(mmrc);
+            Ok(())
+        })?
     }
 }
 
-fn validate_pledge_account<BS, RT>(rt: &RT, addr: &Address) -> Result<(), ActorError>
+////////////////////////////////////////////////////////////////////////////////
+// Method utility functions
+////////////////////////////////////////////////////////////////////////////////
+
+fn compute_initial_pledge<BS, RT>(
+    rt: &mut RT,
+    desc: &SectorStorageWeightDesc,
+) -> Result<TokenAmount, ActorError>
 where
     BS: BlockStore,
     RT: Runtime<BS>,
 {
-    let code_id = rt.get_actor_code_cid(addr)?;
-    if code_id != *MINER_ACTOR_CODE_ID {
-        Err(ActorError::new(
-            ExitCode::ErrIllegalArgument,
-            format!(
-                "pledge account {} must be address of miner actor, was {}",
-                addr, code_id
-            ),
-        ))
-    } else {
-        Ok(())
-    }
+    let st: State = rt.state()?;
+    let ret = rt.send(
+        &*REWARD_ACTOR_ADDR,
+        3,
```

Comment:
I'm so sorry for not being clear in last review, I liked having the actual method in here, I was just suggesting putting on this line instead of binding to a variable
```suggestion
        RewardMethod::LastPerEpochReward as u64,
```
Something like that